src/features/posts.ts

```diff
@@ -0,0 +1,45 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const getPosts = createAsyncThunk<Post[], User['id']>(
+  'posts/fetch',
+  userId => getUserPosts(userId),
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(getPosts.fulfilled, (state, action) => {
+        return {
+          ...state,
+          posts: action.payload,
+        };
+      })
+      .addCase(getPosts.pending, state => {
+        return { ...state, loaded: false };
+      })
+      .addCase(getPosts.rejected, state => {
+        return { ...state, hasError: true };
```

Comment:
When `getPosts` is rejected, you're setting `hasError` to `true`, but you're not resetting it back to `false` when the posts are successfully fetched. This could lead to a situation where `hasError` remains `true` even after a successful fetch. Consider resetting `hasError` in the `getPosts.fulfilled` and `getPosts.pending` cases.